llvm/utils/TableGen/X86RecognizableInstr.cpp

```diff
@@ -223,6 +227,10 @@ InstructionContext RecognizableInstr::insnContext() const {
           insnContext = EVEX_KB_U(IC_EVEX_XD);
         else if (OpPrefix == X86Local::PS)
           insnContext = EVEX_KB_U(IC_EVEX);
+        else {
+          errs() << "Instruction does not use a prefix: " << Name << "\n";
+          llvm_unreachable("Invalid prefix");
```

Comment:
The default op prefix for `EVEX` is `PS` in compiler. 6e20df1a3b0f7654b2821fe182c7ae9bd52672e6

Should we use the message like

```
llvm_unreachable(("Invalid prefix for" + Name).c_str())
```

Reply:
Do you mean remove the `errs() << ...`? How about the others? Both VEX and EVEX have the 4 prefixes.

Reply:
I meant there is always a default prefix opcode `PS` for EVEX prefix, so `does not use a prefix` is misleading.

Yes, I suggest merging the `errs() << ` into `llvm_unreachable`.

Reply:
Oh, I've asked myself some time ago if there is a difference between `llvm_unreachable` and `errs()`. The deal is in production build. The message from `llvm_unreachable` is not shown.

Reply:
I think if the `errs()` is followed by `llvm_unreachable`, the message will be optimized away too in prod build.

Reply:
No sure about the iostream operations, but I remember `llvm_unreachable` won't optimizate function call like printf.

Reply:
So I lean toward to keep both.

Reply:
Update the error message?

Reply:
There are only 2 out of 4141 `grep -rwn llvm_unreachable llvm/lib/ | wc` use `+` to show more detail. I think `llvm_unreachable` is mainly used to show the position to developer, so no need detailed information.